### PR TITLE
#161710003 Delete an Article

### PR DESCRIPTION
**Pull request by brucemakallan**
#### What does this PR do?
Delete a specific Article by id

#### Description of Task to be completed?
- Delete an article by id
- Only the Author can delete the Article

#### How should this be manually tested?
DELETE `/api/articles/<id>`

#### Any background context you want to provide?
You must be authorized to delete an article you created. JWT Token required

#### What are the relevant pivotal tracker stories?
#161710003
https://www.pivotaltracker.com/story/show/161710003

### CODE REVIEW

**Review by RachaelNantale — COMMENTED** [review]
Please work on the merge conflicts

File: authors/apps/core/utils/user_management.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import jwt
```
Comment (emugaya) [inline authors/apps/core/utils/user_management.py:1]: @brucemakallan We already have code doing this. Please confirm

**Review by emugaya — CHANGES_REQUESTED** [review]

File: authors/apps/core/utils/user_management.py (hunk 1 above)
Comment (brucemakallan) [reply authors/apps/core/utils/user_management.py:1]: Retrieving content from a token was used in many different areas so I refactored it to be in just one place I can call it from. It's the method that uses the `import jwt`

**Review by emugaya — APPROVED** [review]

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -98,6 +99,16 @@ def update(self, request, *args, **kwargs):
         serializer.update(article, fresh_article_data)
         return Response(serializer.data, status=status.HTTP_200_OK)
 
+    def destroy(self, request, *args, **kwargs):
+        if not Article.objects.filter(pk=kwargs['pk']).exists():
+            raise APIException({"error": "Article Not found"})
```
Comment (emugaya) [inline authors/apps/articles/views.py]: Raise a NotFound exception

File: authors/apps/articles/serializers.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def to_representation(self, data):
                 "username": user_details.username,
                 "email": user_details.email}
             return article_details
-        raise APIException({"error": "User does not exist"})
+        raise NotFound(detail="User does not exist", code=404)
```
Comment (judeinno) [inline authors/apps/articles/serializers.py:26]: Great job let me also take that on in my work immediately.

**Review by judeinno — APPROVED** [review]

**Review by RachaelNantale — APPROVED** [review]
good work on the exceptions

File: authors/apps/articles/views.py (hunk 2 above)
Comment (emugaya) [reply authors/apps/articles/views.py]: @brucemakallan Why do we still have API Exception here. Make sure that your tests are checking for these status codes.

**Review by dnuwa — APPROVED** [review]

**Review by Sulaiman-Mozes — APPROVED** [review]

File: authors/apps/articles/views.py (hunk 2 above)
Comment (brucemakallan) [reply authors/apps/articles/views.py]: This is now tested for

**Pull request merged**